test/Microsoft.AspNetCore.Server.KestrelTests/ConnectionAdapterTests.cs

```diff
@@ -109,6 +108,54 @@ public class ConnectionAdapterTests
             }
         }
 
+        [Fact]
+        public async Task ThrowingOnReadDoesNotCauseUnobservedException()
+        {
+            var unobservedExceptionThrown = false;
+
+            EventHandler<UnobservedTaskExceptionEventArgs> unobservedExceptionHandler = (sender, args) =>
+            {
+                unobservedExceptionThrown = true;
+            };
+
+            try
+            {
+                TaskScheduler.UnobservedTaskException += unobservedExceptionHandler;
```

Comment:
True, but after the PR there shouldn't be any 😄

Obviously there's a risk that other test code could do this though. Would you recommend reporting exception details so we could track down any unobserved tasks this happens to catch? Only fail if it's the specific exception instance we threw? Or just remove the test?